hack/ci-e2e.sh

```diff
@@ -26,90 +27,98 @@ export PATH="${PATH}:/usr/local/go/bin"
-IMG=quay.io/metal3-io/baremetal-operator:e2e make docker
-
-# Set up minikube
-minikube start --driver=kvm2
-
-virsh -c qemu:///system net-define "${REPO_ROOT}/hack/e2e/net.xml"
-virsh -c qemu:///system net-start baremetal-e2e
-# Attach baremetal-e2e interface to minikube with specific mac.
-# This will give minikube a known reserved IP address that we can use for Ironic
-virsh -c qemu:///system attach-interface --domain minikube --mac="52:54:00:6c:3c:01" \
-  --model virtio --source baremetal-e2e --type network --config
-
-# Restart minikube to apply the changes
-minikube stop
-minikube start
-
-# Load the BMO e2e image into it
-minikube image load quay.io/metal3-io/baremetal-operator:e2e
-
-# Create libvirt domain
-VM_NAME="bmo-e2e-0"
-export BOOT_MAC_ADDRESS="00:60:2f:31:81:01"
-SERIAL_LOG_PATH="/var/log/libvirt/qemu/${VM_NAME}-serial0.log"
-
-virt-install \
-  --connect qemu:///system \
-  --name "${VM_NAME}" \
-  --description "Virtualized BareMetalHost" \
-  --osinfo=ubuntu-lts-latest \
-  --ram=4096 \
-  --vcpus=2 \
-  --disk size=20 \
-  --graphics=none \
-  --console pty,target_type=serial \
-  --serial file,path="${SERIAL_LOG_PATH}" \
-  --xml "./devices/serial/@type=pty" \
-  --xml "./devices/serial/log/@file=${SERIAL_LOG_PATH}" \
-  --xml "./devices/serial/log/@append=on" \
-  --pxe \
-  --network network=baremetal-e2e,mac="${BOOT_MAC_ADDRESS}" \
-  --noautoconsole
-
-# This IP is defined by the network we created above.
-IP_ADDRESS="192.168.222.1"
+IMG="${BMO_IMAGE}" make docker
 
 # These variables are used by the tests. They override variables in the config file.
 # These are the VBMC defaults (used since we did not specify anything else for `vbmc add`).
 export BMC_USER=admin
 export BMC_PASSWORD=password
 
-if [[ "${BMO_E2E_EMULATOR}" == "vbmc" ]]; then
-	# VBMC variables
-  VBMC_PORT="16230"
-  export BMC_ADDRESS="ipmi://${IP_ADDRESS}:${VBMC_PORT}"
 
+# Set up minikube
+profiles=("bmo-e2e" "bmo-e2e-upgrade")
+kubeconfigs=("${HOME}/.kube/config" "/tmp/kubeconfig-e2e-upgrade")
+
+for idx in "${!profiles[@]}"; do
+  profile="${profiles[$idx]}"
+  kubeconfig="${kubeconfigs[$idx]}"
+  virsh -c qemu:///system net-define "${REPO_ROOT}/hack/e2e/${profile}-net.xml"
+  virsh -c qemu:///system net-start "${profile}"
+
+  KUBECONFIG="${kubeconfig}" minikube start --profile "${profile}" --driver=kvm2
+
+  # Attach baremetal-e2e interface to minikube with specific mac.
+  # This will give minikube a known reserved IP address that we can use for Ironic
+  virsh -c qemu:///system attach-interface --domain "${profile}" --mac="52:54:00:6c:3c:0$(( idx+1 ))"\
+    --model virtio --source "${profile}" --type network --config
+
+  # Restart minikube to apply the changes
+  minikube stop --profile "${profile}"
+  KUBECONFIG="${kubeconfig}" minikube start --profile "${profile}"
+
+  # Load the local BMO image into it
+  minikube --profile "${profile}" image load "${BMO_IMAGE}"
+
+  VM_NAME="bmo-e2e-${idx}"
```

Comment:
Could we have a separate name for the VM used for the upgrade? Now they look so similar so it is easy to think that both are used with all tests.

Reply:
Sure. I think that's a good idea.